public/app/plugins/datasource/prometheus/datasource.ts

```diff
@@ -179,29 +179,39 @@ export class PrometheusDatasource extends DataSourceApi<PromQuery, PromOptions>
     const activeTargets: PromQuery[] = [];
 
     for (const target of options.targets) {
-      if (!target.expr || target.hide) {
+      if (!target.expr || target.hide || (target.showingGraph === false && target.showingTable === false)) {
```

Comment:
feels like this condition should be handled in explore and not in the data source :) 

the hide check is also not needed anymore (handled in PanelQueryRunner)

Reply:
Yes, I'll remove that